packages/studio-ui/src/web/views/CodeEditor/SidePanel.tsx

```diff
@@ -0,0 +1,360 @@
+import { Icon } from '@blueprintjs/core'
+import { MainLayout, lang, ModuleUI } from 'botpress/shared'
+import { FileType, HOOK_SIGNATURES } from 'common/code-editor'
+import _ from 'lodash'
+import { inject, observer } from 'mobx-react'
+import React from 'react'
+
+import FileStatus from './components/FileStatus'
+import NameModal from './components/NameModal'
+import NewFileModal from './components/NewFileModal'
+import { UploadModal } from './components/UploadModal'
+import FileNavigator from './FileNavigator'
+import { RootStore, StoreDef } from './store'
+import { EditorStore } from './store/editor'
+import { EXAMPLE_FOLDER_LABEL } from './utils/tree'
+
+const { SearchBar, SidePanel, SidePanelSection } = ModuleUI
+
+class PanelContent extends React.Component<Props> {
+  private expandedNodes = {}
+
+  state = {
+    actionFiles: [],
+    hookFiles: [],
+    botConfigs: [],
+    moduleConfigFiles: [],
+    rawFiles: [],
+    sharedLibs: [],
+    components: [],
+    selectedNode: '',
+    selectedFile: undefined,
+    isMoveModalOpen: false,
+    isCreateModalOpen: false,
+    isUploadModalOpen: false,
+    isComponent: false,
+    fileType: undefined,
+    hookType: undefined
+  }
+
+  componentDidMount() {
+    this.updateFolders()
+  }
+
+  componentDidUpdate(prevProps: Props) {
+    if (prevProps.files !== this.props.files) {
+      this.updateFolders()
+    }
+  }
+
+  addFiles(fileType: string, label: string, fileList: any[]) {
+    const files = this.props.files[fileType]
+
+    if (files && files.length) {
+      const sortedFiles = _.sortBy(files, 'location')
+      fileList.push({ label, files: sortedFiles })
+    }
+  }
+
+  updateFolders() {
+    if (!this.props.files) {
+      return
+    }
+
+    const actionFiles = []
+    this.addFiles('bot.actions', lang.tr('code-editor.sidePanel.currentBot'), actionFiles)
+
+    const hookFiles = []
+    this.addFiles('bot.hooks', lang.tr('code-editor.sidePanel.currentBot'), hookFiles)
+
+    const botConfigFiles = []
+    this.addFiles('bot.bot_config', lang.tr('code-editor.sidePanel.currentBot'), botConfigFiles)
+
+    const moduleConfigFiles = []
+    this.addFiles('bot.module_config', lang.tr('code-editor.sidePanel.currentBot'), moduleConfigFiles)
+
+    const sharedLibs = []
```

Comment:
libs ?

Reply:
Libraries are bot-scoped and bundled with the bot, yes